### ISSUE TITLE
Code Comments

### ISSUE THREAD

**Issue by celsomilne** [ChengF-Lab/GPSnet#2]
I'm finding it quite hard to follow what the code does, and exactly where the code relates to the methodologies described in the paper. I feel that, perhaps, some code comments would assist in the comprehension of the program. 

I'm happy to try and implement this myself. It may be a good precursor to solving #1 .

**Comment by gates-jl** [comment]
@celsomilne - did you make any comments on this code? I'd really appreciate seeing them if possible. Thanks.
